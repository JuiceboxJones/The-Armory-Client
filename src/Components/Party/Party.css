.squad-container {
  position: relative;
  display: flex;
  height: 118px;
  border: 1px solid transparent;
}

.squad-container:hover, .squad-container:focus {
  background-color: #3b3e48 !important;
  border-radius: 0;
  border: 1px solid #6e6f72;
  box-shadow: 0 5px 13px 0 #00000080;
  cursor: pointer;
  z-index: 1;
}

.squad-list {
  margin-top: 124px;
  padding: 0 30px;
  list-style: none;
}

.squad-container {
  margin: 0 -11px;
  border-radius: 5px;
  padding: 0 11px;
}

.squad-container:nth-child(even) {
  background-color: #2c2e35;
}

.squad-list h4 {
  min-width: 160px;
  color: #7faf23;
  font-size: 18px;
  font-weight: 600;
  margin: 0 10px 0 0;

  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.squad-list p {
  margin: 0;
  font-weight: normal;
  color: #b3b3b3;
}

.squad__gamemode-image {
  width: 60px;
  height: 60px;
  box-shadow: 0 1px 8px 0 #00000080;
  border-radius: 5px;
  border: 1px solid #909193;
}

.squad__details {
  display: flex;
  width: 100%;
  flex-direction: column;
  justify-content: center;
}

.squad__details-top {
  display: flex;
  align-items: center;
  margin-bottom: 10px;
}

.squad__spots-list {
  width: 320px;
  list-style: none;
  padding: 0;
}

.squad__spots-list li {
  display: inline-block;
}

.squad__spots-list li:not(:last-child) {
  margin-right: 4px;
}

.squad__spots-full {
  position: absolute;
  top: -9px;
  left: -10px;
  width: 54px;
  height: 60px;
  clip-path: inset(9px 10px 9px 10px);
}

.squad__spots-filled, .squad__spots-empty{
  display: flex;
  position: absolute;
  width: 100%;
  height: 100%;
  justify-content: center;
  align-items: center;
}

.squad__spots-empty {
  background-color: #565d69;
}

.squad__spots-filled {
  background-color: #2c2e35b9;
  color: #fff !important;
}

.squad__spots-image-container {
  position: relative;
  width: 32px;
  height: 38px;
  overflow: hidden;
  border-radius: 5px;
  box-shadow: 0 1px 8px 0 #00000080;
  border: 1px solid #909193;
}

.squad__spots-first-half, .squad__spots-second-half {
  position: absolute;
  display: inline-block;
  font-size: 0;
  width: 52px;
  height: 60px;
  top: -9px;
}

.squad__spots-first-half {
  clip-path: inset(9px 32px 9px 5px);
  left: -5px;
}

.squad__spots-second-half {
  clip-path: inset(9px 5px 9px 32px);
  right: -5px;
}

.squad__gamemode-image-container {
  display: flex;
  align-items: center;
  padding: 0 10px;
}

.squad__requirements {
  display: flex;
  align-items: center;
  font-weight: normal;
}

.squad__avatar-image {
  width: 52px;
  height: 60px;
  position: absolute;
  top: -9px;
  left: -10px;
}

.squad__owner {
  display: flex;
  align-items: center;
  width: 200px;
  padding-bottom: 5px;
}

.squad__owner p {
  color: #fff;
}

.squad__owner div {
  margin-right: 10px;
}