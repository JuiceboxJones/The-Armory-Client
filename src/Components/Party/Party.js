import React from 'react';
import Spot from '../Spot/Spot';
import './Party.css';
import config from '../../config';

export default function Squad(props) {
  function generateSpots(party) {
    return party.spots
      .map((spot, index) => {
        return (
          <Spot key={index} index={index} spot={spot} gameId={props.gameId} partyId={party.id} history={props.history}/>
        );
      });
  }

  function generateRequirements() {
    return props.party.reqs.map((req, i) => {
      if (req.name) {
        return <span className="small-detail" key={i}>{req.name}</span>
      }
    });
  }

  return (
    <li className="squad-container" aria-label='Open Squad Details' tabIndex="0">
      <div className="squad__gamemode-image-container">
        <img className="squad__gamemode-image" src={`${config.IMAGES_ENDPOINT}/${props.gameId}/${props.party.gamemode.icon_url}`} alt=""/>
      </div>
      <div className="squad__details">
        <div className="squad__details-top">
          <h4>{props.party.title}</h4>
          <ul className="squad__spots-list">
            {generateSpots(props.party)}
          </ul>
          <div className="squad__owner">
            <div className="squad__spots-image-container">
              <img className="squad__avatar-image" src={props.party.owner_id.avatar_url} alt=""/>
            </div>
            <p>{props.party.owner_id.username}</p>
          </div>
        </div>
        <p><span className="small-detail">{props.party.gamemode.name}</span>{props.party.description}</p>
      </div>
      <div className="squad__requirements">
        {generateRequirements()}
      </div>
    </li>
  )
}