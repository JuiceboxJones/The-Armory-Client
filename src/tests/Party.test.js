import React from "react";
import Party from "../Components/Party/Party";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

const props = {
  party: {
    id: "fb1d3c63-6a72-4013-be82-5b523c1dd1cd",
    game_id: "aa0e8ce9-1a71-42e7-804d-6838556fa6ed",
    title: "Test Title",
    owner_id: 1,
    description: "This is a description of this testing party.",
    require_app: true,
    gamemode: {
      name: "Quick Play",
      icon_url: ""
    }
  },
  reqs: [
    {
        name: "test reqs 1"
    },
    {
        name: "test reqs 2"
    }
],
spots: [
    {
        id: "64ed5ba8-78db-44c6-ae60-46e6a2a07ff9",
        filled: null,
        roles: [
            {
                name: "test2",
                icon_url: ""
            },
            {
                name: "test2",
                icon_url: ""
            }
        ]
    },
    {
        id: "25539899-aae0-469e-92c1-a2116badc84c",
        filled: {
            "username": "admin",
            "avatar_url": "https://i.ebayimg.com/images/g/PfAAAOSwA3dYIPRN/s-l300.jpg"
        },
        roles: [
            {}
        ]
    }
]
};

describe.skip("Party component", () => {
  console.log(props.party)
  it("renders without crashing", () => {
    shallow(<Party party={props} />);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Party party={props} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
