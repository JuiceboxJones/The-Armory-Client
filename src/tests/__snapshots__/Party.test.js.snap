// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Party component renders the UI as expected 1`] = `
<li
  aria-label="Open Squad Details"
  className="squad-container"
  tabIndex="0"
>
  <div
    className="squad__gamemode-image-container"
  >
    <img
      alt=""
      className="squad__gamemode-image"
      src="http://localhost:8000/images/undefined/Gamemode-Quickplay.png"
    />
  </div>
  <div
    className="squad__details"
  >
    <div
      className="squad__details-top"
    >
      <h4>
        Test Title
      </h4>
      <ul
        className="squad__spots-list"
      >
        <li
          className="squad__spots-image-container"
        >
          <img
            alt=""
            className="squad__spots-first-half"
            src="http://localhost:8000/images/undefined/Icon-Genji.png"
          />
          <img
            alt=""
            className="squad__spots-second-half"
            src="http://localhost:8000/images/undefined/Icon-Support.png"
          />
        </li>
        <li
          className="squad__spots-image-container"
        >
          <i
            className="squad__spots-empty fas fa-users"
          />
          <i
            className=" squad__spots-filled fas fa-check"
          />
        </li>
      </ul>
      <div
        className="squad__owner"
      >
        <div
          className="squad__spots-image-container"
        >
          <img
            alt=""
            className="squad__avatar-image"
          />
        </div>
        <p />
      </div>
    </div>
    <p>
      <span
        className="small-detail"
      >
        Quick Play
      </span>
      This is a description of test party.
    </p>
  </div>
  <div
    className="squad__requirements"
  >
    <span
      className="small-detail"
    >
      Diamond
    </span>
    <span
      className="small-detail"
    >
      Bronze
    </span>
  </div>
</li>
`;
